src/phoenix/server/api/mutations/dataset_mutations.py

```diff
@@ -265,6 +266,88 @@ async def add_examples_to_dataset(
             )
         )
 
+    @strawberry.mutation
+    async def delete_dataset_examples(
+        self, info: Info[Context, None], input: DeleteDatasetExamplesInput
+    ) -> DatasetMutationPayload:
+        example_db_ids = [
+            from_global_id_with_expected_type(global_id, models.DatasetExample.__name__)
+            for global_id in input.example_ids
+        ]
+        # Guard against empty input
+        if not example_db_ids:
+            raise ValueError("Must provide examples to delete")
+        dataset_version_description = (
+            input.dataset_version_description
+            if isinstance(input.dataset_version_description, str)
+            else None
+        )
+        dataset_version_metadata = input.dataset_version_metadata or {}
+        async with info.context.db() as session:
+            # Check if the examples are from a single dataset
+            dataset_ids = (
+                (
+                    await session.execute(
+                        select(models.DatasetExample.dataset_id)
```

Comment:
We could also just pull in the `models.Dataset` with `limit 2`. This will save us one more query at the end
```suggestion
                        select(models.Dataset)
```

Reply:
Oh yeah that's much cleaner thanks